docs/sources/fqdn-templating.md

```diff
@@ -0,0 +1,280 @@
+  - --exclude-domains=invalid.example.com
+```
+
+### Using Annotations for FQDN Templating]
+
+This example demonstrates how to use annotations in Kubernetes objects to dynamically generate Fully Qualified Domain Names (FQDNs) using the --fqdn-template flag in ExternalDNS.
+
+The Service object includes an annotation dns.company.com/label with the value my-org-tld-v2. This annotation is used as part of the FQDN template to construct the DNS name.
+
+```yml
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-v2
+  namespace: my-namespace
+  annotations:
+    dns.company.com/label: my-org-tld-v2
+spec:
+  type: ClusterIP
+  clusterIP: None
+```
+
+The --fqdn-template flag is configured to use the annotation value (dns.company.com/label) and append the namespace and a custom domain (company.local) to generate the FQDN.
+
+```yml
+args:
+  --source=service
+  --fqdn-template='{{index .ObjectMeta.Annotations "dns.company.com/label"}}.{{ .Namespace }}.company.local'
+
+# For the given Service object, the resulting FQDN will be:
+# route53> my-org-tld-v2.my-namespace.company.local
+```
+
+### DNS Scheme Migration
+
+If you're transitioning from one naming convention to another (e.g., from svc.cluster.local to svc.example.com), --fqdn-template allows you to generate the new records alongside or in place of the old ones — without requiring changes to your Kubernetes manifests.
+
+```yml
+args:
+- --fqdn-template='{{.Name}}.new-dns.example.com'
+```
+
+This helps automate DNS record migration while maintaining service continuity.
+
+### Multi-Variant Domain Support
+
+You can also support regional variants or multi-tenant architectures, where the same service is deployed to different regions or environments:
+
+```yaml
+--fqdn-template='{{ .Name }}.{{ .Environment }}.{{.Region}}.example.com, {{ if eq .Environment "prod" }}{{ .Name }}.my-company.tld{{ end }}'
+```
+
+With additional context (e.g., annotations), this can produce FQDNs like:
+
+```yml
+api.dev.us-east-1.example.com
+api.my-company.tld
+```
+
+This is helpful in scenarios such as:
+
+- Blue/green deployments across domains
+- Staging vs. production resolution
+- Multi-cloud or multi-region failover strategies
+
+## Tips
+
+- If `--fqdn-template` is specified, ExternalDNS ignores any `external-dns.alpha.kubernetes.io/hostname` annotations.
+- You must still ensure the resulting FQDN is valid and unique.
+- Since Go templates can be error-prone, test your template with simple examples before deploying. Mismatched field names or nil values (e.g., missing labels) will result in errors or skipped entries.
+
+## FaQ
+
+### Can I specify multiple global FQDN templates?
+
+No, you can. Pass in a comma separated list to --fqdn-template. Beaware this will double (triple, etc) the amount of DNS entries based on how many services, ingresses and so on you have and will get you faster towards the API request limit of your DNS provider.
```

Comment:
```suggestion
Yes, you can. Pass in a comma separated list to --fqdn-template. Beware this will double (triple, etc) the amount of DNS entries based on how many services, ingresses and so on you have and will get you faster towards the API request limit of your DNS provider.
```